### PR TITLE
Morse decode

### PR DESCRIPTION
**Pull request by lisandroseia**
# Main Point 🥇 

🦅 
> In this new PR we created a Ruby project that allows the user to decode a morse code

## Project Schedule 📄 
- Add a ruby method to decode a single character
- Add a ruby method to decode a single word
- Add a ruby method to decode a whole text

## Set up 💻 
- Clone the PR repo Branch https://github.com/lisandroseia/decode-morse.git
- Run `ruby decode-morse`

Thanks for the review, any optional suggestions will be received

### CODE REVIEW

File: morse_decoder.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+end
+
+def decode(text)
+  new_word = ''
+  new_arr = text.split('   ')
+  new_arr.each do |value|
+    new_word += "#{decode_word(value)} "
+  end
+  puts new_word
+end
```
Comment (FrederickMih) [inline morse_decoder.rb]: - [ ] Hi team, nice job so far. Your `decode_char` and `decode_word` methods work as expected however, your decode method does not return the requested result, that's `"MY NAME"`. It rather returns `MY NAME `, next `nil` as seen on the screenshot below: 
Screenshot from 2022-03-30 23-01-40

Please let's fix this issue.

- [ ] Also, let's make sure that the final result should be exactly the same as:
 
```
"A"
"MY"
"MY NAME"
"A BOX FULL OF RUBIES"
```
And **NOT**

"A"
"MY"
MY NAME 
nil
MY NAME 
A BOX FULL OF RUBIES 
nil
A BOX FULL OF RUBIES  

Thanks :pray:

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



# To Highlights

- Git/GitHub workflow has been followed :heavy_check_mark: 
- Linters are correctly set up :heavy_check_mark: 
- Pull Request has a good title and summary :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: morse_decoder.rb (hunk 1 above)
Comment (lisandroseia) [reply morse_decoder.rb]: Hi, I tried a lot of times with different outputs but I never got the 'nil', do you know what could be the problem?

Comment (Nicolaswg) [reply morse_decoder.rb]: morse_decode

Hi @FrederickMih, this is the output that we get from our project.

Comment (lisandroseia) [reply morse_decoder.rb]: Screenshot from 2022-03-30 21-20-18

As you can see, is also the same on my computer, kindly answer so we could get this cleared

File: morse_decoder.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  new_word
+end
+
+def decode(text)
+  new_word = ''
+  new_arr = text.split('   ')
+  new_arr.each do |value|
+    new_word += "#{decode_word(value)} "
+  end
+  puts new_word unless new_word.nil?
```
Comment (mikethreels) [inline morse_decoder.rb]: - [ ] the reason your application is returning `nil` after every sentence is that there is a `puts` statement in line 51 of your code, please remove it.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: morse_decoder.rb (hunk 2 above)
Comment (lisandroseia) [reply morse_decoder.rb]: Hi, @mikethreels, thanks for the explanation, we should just return the value? we thought it should be printed, that's why we used the puts statement

**Comment by mikethreels** [comment]
Hi Lisandro,
Yes just return the value it will be printed by the print statement in front of the method

File: morse_decoder.rb (hunk 1 above)
Comment (FrederickMih) [reply morse_decoder.rb]: Hi, @lisandroseia  sorry for the late reply. If you check the screenshot in your README file, you will discover that your output contains a nill value also.

Comment (lisandroseia) [reply morse_decoder.rb]: Hi, yeah, that's for the IRB, which is always returning nil when we run a "puts", but in our Vscode it was working well so we weren't able to realize how to solve the problem, thanks for taking the time to answer anyway, I appreciate it!

Comment (FrederickMih) [reply morse_decoder.rb]: Hi @Nicolaswg, if you check keenly the output result in your project description, you will discover that the output of the text **A BOX FULL OF RUBIES** is in quotes. But this is not the same result in your output

**Review by FrederickMih — APPROVED** [review]
# Status: Approved :tada: :tada: :champagne: :champagne: 

Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: morse_decoder.rb (hunk 1 above)
Comment (FrederickMih) [reply morse_decoder.rb]: Hi @lisandroseia :smile: Wish you all the best.

**Pull request merged**